src/sexp.ml

```diff
@@ -5,9 +5,9 @@ include (Usexp : module type of struct include Usexp end
 
 let code_error message vars =
   code_errorf "%a" pp
-    (List (Atom message
+    (List (Quoted_string message
            :: List.map vars ~f:(fun (name, value) ->
-             List [Atom name; value])))
+             List [Atom (Atom.of_string name); value])))
```

Comment:
The name is not guaranteed to be an atom. It is in general, but it can be whatever we think will be useful for debugging